### PR TITLE
[FIX] Take account_ids into account on account static balance

### PR DESCRIPTION
**Pull request by rconjour**
[FIX] Take account_ids into account on account static balance

### CODE REVIEW

**Comment by StefanRijnhart** [comment]
Is this a bugfix or rather an optimization?

**Comment by rconjour** [comment]
> Is this a bugfix or rather an optimization?

This is a bugfix for the bug you noted on bitbucket (this error:
```QWebException: "1074" while evaluating'get_lines(data)'```

**Comment by StefanRijnhart** [comment]
Ah, I see now that we prepopulate the `res` dictionary with the extended set of accounts, and if we don't apply a filter on those accounts consistently, the keyerror is raised when the dictionary is queried for an account that was not in that original set.
Thanks!

**Review by StefanRijnhart — APPROVED** [review]

File: account_financial_report_webkit/models/account_static_balance.py [hunk 1]
```diff
@@ -289,8 +289,12 @@ def map_data_to_account_id(data):
                     AND asb.create_date > ajp.write_date
                     AND ajp.state = 'done')
                 AND asb.period_id IN %(period_ids)s
+                AND asb.account_id IN %(account_ids)s
             GROUP BY account_id
-            """, {'period_ids': tuple(periods.ids)})
+            """, {
+                'period_ids': tuple(periods.ids),
+                'account_ids': tuple(accounts.ids)
```
Comment (PieterPaulussen) [inline account_financial_report_webkit/models/account_static_balance.py]: Won't this lead to very large 'IN' array checks?

**Review by StefanRijnhart — APPROVED** [review]

**Review by PieterPaulussen — APPROVED** [review]
Nice, better solution than the ids IN solution!

**Pull request merged**